service/storage_service.cc

```diff
@@ -1620,7 +1752,11 @@ future<> storage_service::init_server(int delay, bind_messaging_port do_bind) {
             if (!tokens.empty()) {
                 _token_metadata.update_normal_tokens(tokens, get_broadcast_address());
                 replicate_to_all_cores().get();
-                // order is important here, the gossiper can fire in between adding these two states.  It's ok to send TOKENS without STATUS, but *not* vice versa.
+
+                // Order is important: our tokens must be known when a node handles our status.
+                // We don't need to send our streams timestamp (we don't know it yet),
+                // because the other nodes won't use it until we leave hibernation.
+                // FIXME? (kbraun): actually, if we're hibernated, we also don't need to send tokens. Noone uses it.
```

Comment:
We are removing the join_ring option which means we also join the ring. So, the code here is going to be removed very soon.

Reply:
Yay!